### PR TITLE
Workflow enhancements

### PR DESCRIPTION
**Pull request by jameshopkins**
- [x] Added ESLint configuration
- [x] Added pre-commit tasks to ensure code quality
- [x] Converted existing codebase to ES6

### CODE REVIEW

File: src/indexed.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {Seq} from 'immutable';
+import Base from './base';
+import {updateCursor} from './utils';
+
+function Indexed(root, data, keyPath, onChange, size) {
```
Comment (charypar) [inline src/indexed.js:5]: May be clearer to use the `class` syntax here...? Unless you're about to rewrite this in a completely functional way.

**Comment by charypar** [comment]
Looks good! :+1:

File: src/indexed.js (hunk 1 above)
Comment (jameshopkins) [inline src/indexed.js:5]: I had pondered about using `class`, however, I can't see how the inheritance pattern I'm using, can fit elegantly into the model that the `class` syntax promotes. For example, in the case of `Indexed` and `Keyed` constructors, I'm inheriting methods from the prototype chains of both `Seq.Indexed` and `Seq.Keyed` respectively, whilst subsequently also 'inheriting' (merging) methods from `Base.prototype` for each. The issue is, `class` and `prototype` promote differential inheritance - which this pattern isn't.

Comment (charypar) [inline src/indexed.js:5]: Do I hear multiple inheritance? :) Would it be a better idea to compose and delegate? 

I don't know exactly what `class` does behind the scenes, but I think it's still prototypal inheritance, just limited to single inheritance. Might be wrong about that.

**Pull request merged**